feat: add support for inline struct creation (#4120)
confluentinc/ksql: docs-md/developer-guide/syntax-reference.md

example, `STRUCT<ID BIGINT, NAME STRING, AGE INT>` defines a struct with
three fields, with the supplied name and type.

You can create a struct in a query by specifying the names of the columns
and expressions that construct the values, separated by `,` and wrapped with
curly braces. For example: `SELECT {name col0, ageInDogYears col1*7} AS dogs FROM animals`

> would create a schema `col0 STRUCT<VARCHAR, INTEGER>` (assuming `col0` was a string and

Reviewer: ```suggestion
creates a schema, `col0 STRUCT<VARCHAR, INTEGER>`, assuming `col0` is a string and
```

> `col1` was an integer).

Reviewer: ```suggestion
`col1` is an integer.
```
